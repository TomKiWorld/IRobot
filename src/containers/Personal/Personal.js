import React from 'react';

import FindInput from '../../components/FindInput/FindInput';
import ErrorBoundry from '../../components/ErrorBoundry/ErrorBoundry';
import PersonalImage from '../../components/PersonalImage/PersonalImage';

const Personal = ({ onNameChange, name }) => {
  const image = name ? <PersonalImage name={name} /> : 
  <main className='light-green container'>
    <p>Please type your name to find your Personal</p>
    <p className='sega'>Robot</p>
  </main>;
  return (
    <main className='container'>
      <FindInput 
        onNameChange={onNameChange}
        name={name}
      />
      <ErrorBoundry>
        {image}
      </ErrorBoundry>
    </main>
  );
};

export default Personal;
