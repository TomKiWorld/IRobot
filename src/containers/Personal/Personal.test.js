import { shallow } from 'enzyme';
import React from 'react';
import Personal from './Personal';

describe('Personal Component', () => {
  it('Expect to render Personal Component', () => {
    expect(shallow(<Personal name={''}/>)).toMatchSnapshot();
  });

  it('Expect to render Personal Component with name', () => {
    expect(shallow(<Personal name={'John'}/>)).toMatchSnapshot();
  });
});
