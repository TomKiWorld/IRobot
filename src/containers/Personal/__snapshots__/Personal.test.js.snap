// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal Component Expect to render Personal Component 1`] = `ShallowWrapper {}`;

exports[`Personal Component Expect to render Personal Component with name 1`] = `ShallowWrapper {}`;
